internal/config/attestation.go

```diff
@@ -16,12 +16,44 @@ import (
 	"github.com/edgelesssys/constellation/v2/internal/variant"
 )
 
-// AttestationConfig is the common interface for passing attestation configs.
-type AttestationConfig interface {
+// AttestationCfg is the common interface for passing attestation configs.
+type AttestationCfg interface {
 	// GetMeasurements returns the measurements that should be used for attestation.
 	GetMeasurements() measurements.M
+	// SetMeasurements updates a config's measurements using the given measurements.
+	SetMeasurements(m measurements.M)
 	// GetVariant returns the variant of the attestation config.
 	GetVariant() variant.Variant
+	// NewerThan returns true if the config is newer than the given config.
+	NewerThan(AttestationCfg) (bool, error)
```

Comment:
While `NewerThan` isn't 100% correct, neither is `Diff`, since the function should return false if the given config tries to downgrade SNP SVNs.
The goal of the function is also not to detect diffs, but to detect if B is newer than A.
While we can't judge if measurements A are newer than measurements B, we can do so for SVNs, hence my naming for the function.

Reply:
Decided to replace `NewerThan` with an `EqualTo` method.
I think this makes things a bit clearer.